pkg/podeffect/pod_effect_test.go

```diff
@@ -0,0 +1,163 @@
+			name:         "node has more terminating pods get higher score",
+		},
+		{
+			nodeInfos:    []*framework.NodeInfo{makeNodeInfo("machine1", 0, 2), makeNodeInfo("machine2", 0, 1)},
+			expectedList: []framework.NodeScore{{Name: "machine1", Score: framework.MinNodeScore}, {Name: "machine2", Score: framework.MaxNodeScore}},
+			name:         "node has more nominated pods get lower score",
+		},
+		{
+			nodeInfos:    []*framework.NodeInfo{makeNodeInfo("machine1", 5, 2), makeNodeInfo("machine2", 3, 1)},
+			expectedList: []framework.NodeScore{{Name: "machine1", Score: framework.MaxNodeScore}, {Name: "machine2", Score: framework.MinNodeScore}},
+			name:         "node has more (terminatingPodNumber - nominatedPodNumber) get higher score",
+		},
+		{
+			nodeInfos:    []*framework.NodeInfo{makeNodeInfo("machine1", 5, 4), makeNodeInfo("machine2", 3, 1)},
+			expectedList: []framework.NodeScore{{Name: "machine1", Score: framework.MinNodeScore}, {Name: "machine2", Score: framework.MaxNodeScore}},
+			name:         "node has less (terminatingPodNumber - nominatedPodNumber) pods get lower score",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cs := clientsetfake.NewSimpleClientset()
+			informerFactory := informers.NewSharedInformerFactory(cs, 0)
+			registeredPlugins := []st.RegisterPluginFunc{
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterPluginAsExtensions(PodEffectName, New, "Score"),
+			}
+			fakeSharedLister := &fakeSharedLister{nodes: test.nodeInfos}
+
+			fh, err := st.NewFramework(
+				registeredPlugins,
+				frameworkruntime.WithClientSet(cs),
+				frameworkruntime.WithInformerFactory(informerFactory),
+				frameworkruntime.WithSnapshotSharedLister(fakeSharedLister),
+				frameworkruntime.WithPodNominator(testutil.NewPodNominator()),
+			)
+			if err != nil {
+				t.Fatalf("fail to create framework: %s", err)
+			}
+			// initialize nominated pod by adding nominated pods into nominatedPodMap
+			for _, n := range test.nodeInfos {
+				for _, p := range n.Pods {
+					if p.Pod.Status.NominatedNodeName != "" {
+						addNominatedPod(p.Pod, n.Node().Name, fh.PreemptHandle())
+					}
+				}
+			}
+			pe, _ := New(nil, fh)
+			var gotList framework.NodeScoreList
+			plugin := pe.(framework.ScorePlugin)
+			for i, n := range test.nodeInfos {
+				score, err := plugin.Score(context.Background(), nil, nil, n.Node().Name)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				gotList = append(gotList, framework.NodeScore{Name: test.nodeInfos[i].Node().Name, Score: score})
+			}
+
+			status := plugin.ScoreExtensions().NormalizeScore(context.Background(), nil, nil, gotList)
+			if !status.IsSuccess() {
+				t.Errorf("unexpected error: %v", status)
+			}
+
+			for i := range gotList {
+				if !reflect.DeepEqual(test.expectedList[i].Score, gotList[i].Score) {
```

Comment:
As you're just comparing `string`, it's good enough to check if the 2 scores are equal.